src/test/regress/sql/multi_truncate.sql

```diff
@@ -177,3 +177,25 @@ TRUNCATE TABLE "a b append";
 SELECT shardid FROM pg_dist_shard where logicalrelid = '"a b append"'::regclass;
 
 DROP TABLE "a b append";
+
+-- Run master_truncate_local_data on a distributed table
+CREATE TABLE test_local_truncate (x int, y int);
+INSERT INTO test_local_truncate VALUES (1,2);
+SELECT create_distributed_table('test_local_truncate', 'x', colocate_with => 'none');
+
+BEGIN;
+SET LOCAL citus.enable_ddl_propagation TO off;
+TRUNCATE test_local_truncate;
+COMMIT;
+
+-- Ensure distributed data is not truncated
+SELECT * FROM test_local_truncate;
+
+-- Undistribute table
```

Comment:
Can you add a test that does BEGIN; TRUNCATE; ROLLBACK; SELECT rows_still_there?